CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/LoggingRequestHandler.java

```diff
@@ -0,0 +1,88 @@
+package com.couchbase.CouchbaseLiteServ.server.RequestHandler;
+
+import android.content.Context;
+
+import com.couchbase.CouchbaseLiteServ.MainActivity;
+import com.couchbase.CouchbaseLiteServ.server.Args;
+import com.couchbase.lite.Database;
+import com.couchbase.lite.LogFileConfiguration;
+import com.couchbase.lite.LogLevel;
+
+import java.io.File;
+
+public class LoggingRequestHandler {
+    /* ----------- */
+    /* - Logging - */
+    /* ----------- */
+
+    public String configure(Args args){
+        String log_level = args.get("log_level");
+        String directory = args.get("directory");
+        int maxRotateCount = args.get("max_rotate_count");
+        long maxSize = args.get("max_size");
+        boolean plainText = args.get("plain_text");
+
+        if (directory.isEmpty()) {
+            Context context = MainActivity.getAppContext();
+            long ts = System.currentTimeMillis()/1000;
+            directory = context.getFilesDir().getAbsolutePath() + "/logs_" + ts ;
+
+            System.out.println("File logging configured at: " + directory.toString());
+        }
+        LogFileConfiguration config = new LogFileConfiguration(directory);
+        if (maxRotateCount > 1) {
+            config.setMaxRotateCount(maxRotateCount);
+        }
+        if (maxSize > 512000) {
+            config.setMaxSize(maxSize);
+        }
+        config.setUsePlaintext(plainText);
+        Database.log.getFile().setConfig(config);
+        switch (log_level) {
+            case "debug":
+                Database.log.getFile().setLevel(LogLevel.DEBUG);
+                break;
+            case "verbose":
+                Database.log.getFile().setLevel(LogLevel.VERBOSE);
+                break;
+            case "info":
+                Database.log.getFile().setLevel(LogLevel.INFO);
+                break;
+            case "error":
+                Database.log.getFile().setLevel(LogLevel.ERROR);
+                break;
+            case "warning":
+                Database.log.getFile().setLevel(LogLevel.WARNING);
+                break;
+            default:
+                Database.log.getFile().setLevel(LogLevel.NONE);
+                break;
+        }
+        return directory;
```

Comment:
instead of returning directory, returning config might make sense, so that u can use it for set methods. You can get the directory by getDirectory method, so no need to return directory here. Something similiar to replicatorConfiguration, databaseConfiguration we used

Reply:
I returned directory, as configuring logging is just one-time scenario where we won't be doing much changes in the configuration and directory is configure at app level with no knowledge to user. Hence I returned directory. However, if you think I should change the signature of method, I'll do it.

Reply:
can you change it? As the method says configure, it is best practice to return config. I understood your point too, just to make it best practice and align with method name and align with other configuration file, it is better to have config returned.

Reply:
done